### ISSUE TITLE
IDE0060 false positive with [EnumeratorCancellation] attribute

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#71719]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/IDE0060-false-positive-with-EnumeratorC/10564313)._

---
[severity:It bothers me. A fix would be nice]
The IDE0060 rule suggests removing the 'token' parameter thinking it is unused. However, since it has the [EnumeratorCancellation] attribute it is used internally by the compiler even though it isn't referenced by the code.

````c#
using System.Runtime.CompilerServices;
namespace AsyncIteratorBug;

public class Class1(int count)
{
    public async IAsyncEnumerable<int> Foo([EnumeratorCancellation] CancellationToken token)
    {
        for (int i = 0; i < count; i++)
        {
            await Task.Yield();
            yield return i;
        }
    }
}
````

---
#### Feedback Bot on 1/18/2024, 10:46 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
We would take a targeted community PR here.

**Comment by CyrusNajmabadi** [comment]
Closing as by design.  It's fine for this parameter to be removed since it really isn't used.  The compiler is totally fine calling such method.  The only rule is that *if* there is a cancellation token, then it *should* be marked as EnumeratorCancellation
